src/ansys/dpf/core/data_sources.py

```diff
@@ -214,30 +301,44 @@ def set_domain_result_file_path(
         else:
             self._api.data_sources_set_domain_result_file_path_utf8(self, str(path), domain_id)
 
-    def add_file_path(self, filepath, key="", is_domain: bool = False, domain_id=0):
-        """Add a file path to the data sources.
+    def add_file_path(
+        self,
+        filepath: Union[str, os.PathLike],
+        key: Optional[str] = "",
+        is_domain: Optional[bool] = False,
+        domain_id: Optional[int] = 0,
```

Comment:
I do not think you have to use `Optional` here as giving them a default value makes them optional already. Plus, using `Optional` technically means it accepts `None` as a value, right? Which is not the case here.

Reply:
I haven't checked previous code for that in particular but it is worth checking for, because otherwise the user can call these with `None` and it will not work.